### PR TITLE
List notable implementation details of the various engines.

### PR DESCRIPTION
**Pull request by jpountz**
It would be interesting for readers of the benchmark results to better understand how the various engines differ. While many readers stop at Rust vs. Java vs. C++, there are lots of other notable implementation differences such as WAND vs. MAXSCORE or whether or not doc ID reordering is used.

I thought it was also interesting to highlight similarities, such as the use of mmap and PFOR-delta.

### CODE REVIEW

**Comment by jpountz** [comment]
Here is what it looks like:

screenshot-impl-details

**Comment by jpountz** [comment]
I made a change to make these changes folded by default, to avoid adding too much noise to the page. Here's what it looks like when folded:

screenshot-impl-details-folded

And unfolded:

screenshot-impl-details

**Comment by fulmicoton** [comment]
tantivy is FOR, not PFOR. (or maybe I misunderstand what PFOR means)

**Comment by jpountz** [comment]
Woops, thanks for noticing.

File: engines/tantivy-0.22/details.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[
+  "Written in Rust.",
+  "Index is read from disk via mmap.",
+  "Doc-at-a-time.",
+  "Doc deltas, frequencies and positions are encoded using FOR-delta and blocks of size 128.",
+  "Single level of skip data inlined into postings, every 128 docs.",
+  "Impacts are inlined into skip data.",
```
Comment (fulmicoton) [inline engines/tantivy-0.22/details.json:7]: I would not hesitate being more critical here.

Comment (fulmicoton) [inline engines/tantivy-0.22/details.json:7]:
```suggestion
  "Impacts are inlined into skip data: as an approximation only the optimal `(term_freq, norm)` pair for the block is stored. This argmax is computed assuming the current segment statistics. During search, these args are reinjected into the BM25 formula with the global statistics.",
```

File: engines/pisa-0.8.2/details.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[
+  "Written in C++.",
+  "Index is read from disk via mmap.",
+  "Doc-at-a-time.",
+  "Doc deltas, frequencies and positions are encoded using Lemire's FastPFOR library with blocks of size 128.",
+  "Implements block-max WAND / block-max AND for top-k commands.",
```
Comment (fulmicoton) [inline engines/pisa-0.8.2/details.json:6]: Impacts are stored directly, and require a single segment.

Comment (fulmicoton) [inline engines/pisa-0.8.2/details.json:6]:
```suggestion
  "Doc deltas, frequencies and positions are encoded using Lemire's FastPFOR library with blocks of size 128.",
  "Implements block-max WAND / block-max AND for top-k commands.",
   "Impacts are stored directly, and require a single segment."
```

**Comment by jpountz** [comment]
@fulmicoton Does it look ready to merge?

**Pull request merged**